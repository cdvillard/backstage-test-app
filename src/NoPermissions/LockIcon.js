// (C) Copyright 2017-2022 Hewlett Packard Enterprise Development LP
import React from 'react';

function LockIcon() {
  return (
    <svg width='71' height='96' fill='none' xmlns='http://www.w3.org/2000/svg'>
      <g clipPath='url(#clip0)'>
        <path d='M58.022 36.416h-3.016V20.8c0-9.808-7.947-17.776-17.729-17.776h-5.521c-9.782 0-17.73 7.968-17.73 17.776v15.408h-3.015V20.8C11.01 9.328 20.314 0 31.756 0h5.521C48.72 0 58.022 9.328 58.022 20.8v15.616z' fill='#000' />
        <path d='M70.054 96H0V35.408h41.314v3.024H3.016v54.544h64.022V38.432H54.942v-3.024h15.112V96z' fill='#000' />
        <path d='M59.394 87.536H10.676V45.264h48.718v42.272zm-45.702-3.024h42.686V48.288H13.692v36.224z' fill='#00E8CF' />
        <path d='M38.889 76.944h-7.676v-8.432a7.216 7.216 0 01-3.367-6.096c0-3.968 3.224-7.2 7.181-7.2s7.181 3.232 7.181 7.2a7.233 7.233 0 01-3.32 6.064v8.464zm-4.66-3.024h1.644v-7.312l.878-.4a4.185 4.185 0 002.441-3.792 4.174 4.174 0 00-4.165-4.176 4.173 4.173 0 00-1.691 7.984l.893.4v7.296z' fill='#FF8300' />
      </g>
      <defs>
        <clipPath id='clip0'>
          <path fill='#fff' d='M0 0h70.054v96H0z' />
        </clipPath>
      </defs>
    </svg>
  );
}

export default LockIcon;
