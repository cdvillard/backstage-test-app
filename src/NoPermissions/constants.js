// (C) Copyright 2017-2022 Hewlett Packard Enterprise Development LP
export const DOCUMENTATION_URL = 'https://www.hpe.com/info/greenlakecentral-content-depends-on-role';
