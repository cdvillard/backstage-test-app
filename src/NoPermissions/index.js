// (C) Copyright 2017-2022 Hewlett Packard Enterprise Development LP

import React from 'react';
import { Box, Text, Anchor } from 'grommet';
import { useIntl } from 'react-intl';
import LockIcon from './LockIcon';
import { DOCUMENTATION_URL } from './constants';

export const NoPermissions = () => {
  const { formatMessage } = useIntl();

  return (
    <Box
      full={true}
    >
      <Box
        align='center'
        flex='grow'
        justify='center'
        data-e2e='no-permissions'
      >
        <Box align='center' gap='medium'>
          <LockIcon />
          <Text>
            {formatMessage({id: 'noPermissions.description'})}
          </Text>
          <Text>
            {formatMessage(
              {id: 'noPermissions.nextSteps'},
              {
                dashboardLink: (
                  <Anchor
                    key='dashboardLink'
                    href='/'
                    label={formatMessage({id: 'noPermissions.dashboard'})}
                  />),
              }
            )}
          </Text>
          <Box direction='row' gap='xsmall'>
            <Anchor
              href={DOCUMENTATION_URL}
              target='_blank'
              rel='noopener noreferrer'
              label={formatMessage({id: 'noPermissions.documentation'})}
            />
          </Box>
        </Box>
      </Box>
    </Box>
  );
}
