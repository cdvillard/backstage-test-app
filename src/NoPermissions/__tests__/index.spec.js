// (C) Copyright 2017-2022 Hewlett Packard Enterprise Development LP

import React from 'react';
import { render } from '@testing-library/react';
import NoPermissions from '..';

describe('app-components/NoPermissions', () => {
  it('matches the snapshot', () => {
    const { container } = render(<NoPermissions />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
