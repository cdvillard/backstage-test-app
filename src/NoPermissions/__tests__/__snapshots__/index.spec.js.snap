// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-components/NoPermissions matches the snapshot 1`] = `
<div
  class="StyledBox-sc-13pk1d4-0 jLMOlN"
>
  <div
    class="StyledBox-sc-13pk1d4-0 bAhRyN"
    data-e2e="no-permissions"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 hXkiLl"
    >
      <svg
        fill="none"
        height="96"
        width="71"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0)"
        >
          <path
            d="M58.022 36.416h-3.016V20.8c0-9.808-7.947-17.776-17.729-17.776h-5.521c-9.782 0-17.73 7.968-17.73 17.776v15.408h-3.015V20.8C11.01 9.328 20.314 0 31.756 0h5.521C48.72 0 58.022 9.328 58.022 20.8v15.616z"
            fill="#000"
          />
          <path
            d="M70.054 96H0V35.408h41.314v3.024H3.016v54.544h64.022V38.432H54.942v-3.024h15.112V96z"
            fill="#000"
          />
          <path
            d="M59.394 87.536H10.676V45.264h48.718v42.272zm-45.702-3.024h42.686V48.288H13.692v36.224z"
            fill="#00E8CF"
          />
          <path
            d="M38.889 76.944h-7.676v-8.432a7.216 7.216 0 01-3.367-6.096c0-3.968 3.224-7.2 7.181-7.2s7.181 3.232 7.181 7.2a7.233 7.233 0 01-3.32 6.064v8.464zm-4.66-3.024h1.644v-7.312l.878-.4a4.185 4.185 0 002.441-3.792 4.174 4.174 0 00-4.165-4.176 4.173 4.173 0 00-1.691 7.984l.893.4v7.296z"
            fill="#FF8300"
          />
        </g>
        <defs>
          <clippath
            id="clip0"
          >
            <path
              d="M0 0h70.054v96H0z"
              fill="#fff"
            />
          </clippath>
        </defs>
      </svg>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 iCeYCB"
      />
      <span
        class="StyledText-sc-1sadyjn-0 gKFELF"
      >
        noPermissions.description
      </span>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 iCeYCB"
      />
      <span
        class="StyledText-sc-1sadyjn-0 gKFELF"
      >
        noPermissions.nextSteps
      </span>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 iCeYCB"
      />
      <div
        class="StyledBox-sc-13pk1d4-0 wrGSV"
      >
        <a
          class="StyledAnchor-sc-1rp7lwl-0 gWhgbM"
          href="https://www.hpe.com/info/greenlakecentral-content-depends-on-role"
          rel="noopener noreferrer"
          target="_blank"
        >
          noPermissions.documentation
        </a>
      </div>
    </div>
  </div>
</div>
`;
